internal/config/api.go

```diff
@@ -66,24 +73,89 @@ func (a *API) init() error {
 	}
 
 	if authInfo == nil || (authInfo.APIKey == "" && authInfo.BearerToken == "") {
-		return errors.New("No authentication found. Please log in using 'signadot auth login'")
+		return AuthNoFoundError
 	}
-	if authInfo.ExpiresAt != nil && authInfo.ExpiresAt.Before(time.Now()) {
-		return errors.New("Authentication expired. Please log in using 'signadot auth login'")
+	if authInfo.ExpiresAt != nil && authInfo.ExpiresAt.Before(time.Now()) && authInfo.Source != auth.KeyringAuthSource {
+		return AuthExpiredError
 	}
 	if authInfo.OrgName == "" {
-		return errors.New("No organisation found. Please log in using 'signadot auth login'")
+		return AuthNoOrgFoundError
+	}
+
+	// Init basic settings and return
+	a.basicInit()
+
+	if authInfo.Source == auth.KeyringAuthSource {
+		if err := a.checkKeyringAuth(authInfo); err != nil {
+			return err
+		}
 	}
 
 	a.ApiKey = authInfo.APIKey
 	a.BearerToken = authInfo.BearerToken
 	a.Org = authInfo.OrgName
+	return nil
+}
+
+func (a *API) checkKeyringAuth(authInfo *auth.ResolvedAuth) error {
+
+	// If the auth is expired, we need to refresh the token
+	if authInfo.ExpiresAt != nil && time.Now().After(*authInfo.ExpiresAt) {
+		if authInfo.RefreshToken == "" {
+			return AuthExpiredError
+		}
+
+		newAuthInfo, err := a.refreshKeyringAuth(authInfo)
+		if err != nil {
+			return err
+		}
+		authInfo = newAuthInfo
```

Comment:
This doesn't do anything, because Go is call by value (see https://go.dev/doc/faq#pass_by_value)

did you mean `*authInfo = *newAuthInfo`?